src/utils/IdealGraphVisualizer/View/src/com/sun/hotspot/igv/view/NodeQuickSearch.java

```diff
@@ -109,25 +109,39 @@ public void evaluate(SearchRequest request, SearchResponse response) {
             if (matches != null) {
                 final Set<InputNode> set = new HashSet<>(matches);
                 final InputGraph theGraph = p.getGraph() != matchGraph ? matchGraph : null;
-                response.addResult(new Runnable() {
-                    @Override
-                    public void run() {
-                        final EditorTopComponent comp = EditorTopComponent.getActive();
-                        if (comp != null) {
-                            if (theGraph != null) {
-                                comp.getDiagramModel().selectGraph(theGraph);
+                // Show "All N matching nodes" entry only if 1) there are
+                // multiple matches and 2) the query does not only contain
+                // digits (it is rare to select all nodes whose id contains a
+                // certain subsequence of digits).
+                if (matches.size() > 1 && !rawValue.matches("\\d+")) {
+                    if (!response.addResult(new Runnable() {
+                        @Override
+                        public void run() {
+                            final EditorTopComponent comp = EditorTopComponent.getActive();
+                            if (comp != null) {
+                                if (theGraph != null) {
+                                    comp.getDiagramModel().selectGraph(theGraph);
+                                }
+                                comp.setSelectedNodes(set);
+                                comp.requestActive();
                             }
-                            comp.setSelectedNodes(set);
-                            comp.requestActive();
                         }
+                    },
+                            "All " + matches.size() + " matching nodes (" + name + "=" + value + ")" + (theGraph != null ? " in " + theGraph.getName() : "")
+                    )) {
+                        return;
                     }
-                },
-                        "All " + matches.size() + " matching nodes (" + name + "=" + value + ")" + (theGraph != null ? " in " + theGraph.getName() : "")
-                );
+                }
+
+                // Rank the matches.
+                Collections.sort(matches,
+                                 (InputNode a, InputNode b) ->
+                                 Integer.valueOf(rankMatch(rawValue, a.getProperties().get(name)))
+                                 .compareTo(rankMatch(rawValue, b.getProperties().get(name))));
```

Comment:
You can use `Integer.compare()` which directly takes two `ints` instead of `compareTo()`.

Reply:
Thanks, done!